src/cmd-cloud-prune

```diff
@@ -0,0 +1,303 @@
+
+import argparse
+import json
+import yaml
+import collections
+import datetime
+import os
+import boto3
+from dateutil.relativedelta import relativedelta
+import requests
+from tenacity import retry, retry_if_exception_type
+
+retry_requests_exception = (
+    retry_if_exception_type(requests.Timeout)
+    | retry_if_exception_type(requests.ReadTimeout)
+    | retry_if_exception_type(requests.ConnectTimeout)
+    | retry_if_exception_type(requests.ConnectionError)
+)
+
+from cosalib.gcp import remove_gcp_image
+from cosalib.aws import deregister_ami, delete_snapshot
+from cosalib.builds import BUILDFILES
+from cosalib.cmdlib import (
+    retry_stop_long,
+    retry_wait_long,
+    retry_boto_exception,
+    retry_callback,
+)
+
+Build = collections.namedtuple("Build", ["id", "timestamp", "images", "arch"])
+# set metadata caching to 5m
+CACHE_MAX_AGE_METADATA = 60 * 5
+
+
+def main():
+    args = parse_args()
+    bucket, prefix = args.url.split("/", 1)
+    stream = args.stream
+    # Boto3 loads credentials from ~/.aws/config by default and we can change
+    # this default location by setting the AWS_CONFIG_FILE environment variable.
+    # The alternative is to manually pass ACCESS_KEY and SECRET_KEY which isn't favourable.
+    if args.aws_config_file:
+        os.environ["AWS_CONFIG_FILE"] = args.aws_config_file
+
+    gcp_cloud_config = {
+        "gcp": {
+            "json-key": args.gcp_json_key,
+            "project": args.gcp_project,
+        }
+    }
+    # These lists are up to date as of schema hash
+    # 4c19aed3b3d84af278780bff63728510bb3e70613e4c4eef8cabd7939eb31bd8. If changing
+    # this hash, ensure that the list of supported and unsupported artifacts below
+    # is up to date.
+    supported = ["amis", "gcp"]
+    unsupported = []
+
+    with open(f"{args.policy}") as f:
+        policy = yaml.safe_load(f)
+
+    if stream not in policy:
+        print(f"No pruning policy specified for {stream} stream in policy.yaml")
+        return
+    else:
+        s3_client = boto3.client("s3")
+        # If the build key is set in the policy file, then the cloud-uploads key must
+        # also be present, and the duration of cloud-uploads must be equal or shorter
+        if "build" in policy[stream].keys():
+            cloud_uploads_check(policy[stream])
+
+        # Base URL for Fedora CoreOS builds
+
+        buildJsonData = get_json_from_s3(bucket, prefix + "/builds/builds.json")
+        if buildJsonData is None:
+            print(f"No builds.json found for {stream} in {bucket} bucket")
+            return
+
+        # action is basically whatever is needed to be pruned for the respective stream
+        for action in policy[stream]:
+            duration = policy[stream][action]
+            duration_in_months = get_months(duration)
+            refDate = datetime.datetime.now() - relativedelta(months=int(duration_in_months))
+
+            print(f"Pruning {duration} old {action} for {stream} builds")
```

Comment:
I think for debuggability, it'd be better to echo back what we parsed and calculated here. E.g.

```suggestion
            print(f"Pruning resources of type {action} older than {duration_in_months} months ({ref_date}) on stream {stream}")
```